src/sbd-inquisitor.c

```diff
@@ -1020,11 +1055,13 @@ int main(int argc, char **argv, char **envp)
 			cl_log(LOG_INFO, "Realtime mode deactivated.");
 			break;
 		case 'S':
-			start_mode = atoi(sanitized_optarg);
+			rc = strtonum(sanitized_optarg, &start_mode, start_mode);
+			CHECK_RC(rc, "Invalid value for 'start mode': %s\n", sanitized_optarg);
```

Comment:
hmm ... if it is an empty string we get the single-letter in the output and for other issues with it e.g. 'start mode' here.
As setting the default anyway isn't effective when we bail out - and we don't check for individualy valid numbers per option - we could maybe stay consistent with the string-stuff and catch it early. Then we don't need the macro and would replace atoi(sanitized_optarg) simply with sanitized_int_optarg.

Reply:
> As setting the default anyway isn't effective when we bail out - and we don't check for individualy valid numbers per option - we could maybe stay consistent with the string-stuff and catch it early. 


Yes actually setting the default value isn't effective as we bail out.
We can skip the bail out thing and just issue a warning for all options ?

Reply:
> hmm ... if it is an empty string we get the single-letter in the output and for other issues with it e.g. 'start mode' here.
> As setting the default anyway isn't effective when we bail out - and we don't check for individualy valid numbers per option - we could maybe stay consistent with the string-stuff and catch it early. Then we don't need the macro and would replace atoi(sanitized_optarg) simply with sanitized_int_optarg.


Hm, sorry I don't get the idea for the > we could maybe stay consistent with the string-stuff and catch it early

Reply:
> > hmm ... if it is an empty string we get the single-letter in the output and for other issues with it e.g. 'start mode' here.
> > As setting the default anyway isn't effective when we bail out - and we don't check for individualy valid numbers per option - we could maybe stay consistent with the string-stuff and catch it early. Then we don't need the macro and would replace atoi(sanitized_optarg) simply with sanitized_int_optarg.
> 
> Hm, sorry I don't get the idea for the > we could maybe stay consistent with the string-stuff and catch it early

Right you mean sanitize `sanitized_optarg` and call `atoi(sanitized_int_optarg)` with the sanitized string ?

Reply:
```
if (optarg && (
    (sanitized_optarg == NULL) ||
    (strchr("...", c) && (sanitized_int_optarg = sanitize_option_int_value(sanitized_optarg)) < 0))) {
...
}
```

Reply:
> ```
> ...
> int sanitized_int_optarg = 0; /* code wouldn't require it but probably coverity likes it to be set */
> ...
> if (optarg && (
>     (sanitized_optarg == NULL) ||
>     (strchr("...", c) && (sanitized_int_optarg = sanitize_option_int_value(sanitized_optarg) ) < 0))) {
> ...
> }
> ```

I see adding a similar function to `sanitize_option_value` just for the numerics options like `sanitize_option_int_value` and catch it early.

Reply:
yep - guess this is the least code intrusive to have a solution for now and then we can think over how it would look like when going for glib-cmdline-parsing and/or a nice more capable sanitizer.